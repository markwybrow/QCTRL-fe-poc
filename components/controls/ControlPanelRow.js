import React from 'react';
import TypePill from '../common/TypePill';
import styled from '@emotion/styled';

const ArrowRight = styled.div`
	img {
		height: 14px;
		width: auto;
	}
`;
const ControlPanelRow = ({ attributes }) => {
	const { name, type, maximum_rabi_rate, polar_angle } = attributes;

	return (
		<li>
			<div className="column__data">{name}</div>
			<div className="column__data">
				<TypePill label={type} />
			</div>
			<div className="column__data">{maximum_rabi_rate}</div>
			<div className="column__data">{polar_angle}</div>
			<div className="column__action">
				<ArrowRight>
					<img src={'/static/assets/arrow-right.svg'} />
				</ArrowRight>
			</div>
		</li>
	);
};
export default ControlPanelRow;
