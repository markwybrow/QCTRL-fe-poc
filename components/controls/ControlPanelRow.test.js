import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import ControlPanelRow from './ControlPanelRow';

const data = {
	type: 'controls',
	id: '1',
	attributes: {
		name: 'Single-Qubit Driven',
		type: 'Primitive',
		maximum_rabi_rate: 63.16731,
		polar_angle: 0.05671,
	},
};

describe('<ControlPanelRow /> spec', () => {
	it(' renders the component with data correctly ', () => {
		const wrapper = render(<ControlPanelRow {...data} />);
		expect(wrapper.firstChild).toMatchSnapshot();
	});
	it(' assert there are div.column__ rendered ', () => {
		const wrapper = shallow(<ControlPanelRow {...data} />);
		expect(wrapper.find('.column__data').length).toBe(4);
		expect(wrapper.find('.column__action').length).toBe(1);
	});
	it(' assert the data is rendered in each column in the correct order ', () => {
		const { name, type, maximum_rabi_rate, polar_angle } = data.attributes;
		const wrapper = mount(<ControlPanelRow {...data} />);
		const divs = wrapper.find('div');
		expect(
			divs
				.find('div')
				.at(0)
				.text(),
		).toEqual(name);
		expect(
			divs
				.find('div')
				.at(1)
				.text(),
		).toEqual(type);
		expect(
			divs
				.find('div')
				.at(4)
				.text(),
		).toEqual(`${maximum_rabi_rate}`);
		expect(
			divs
				.find('div')
				.at(5)
				.text(),
		).toEqual(`${polar_angle}`);
	});
});
