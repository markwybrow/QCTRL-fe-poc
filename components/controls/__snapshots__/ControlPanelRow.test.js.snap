// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlPanelRow /> spec  renders the component with data correctly  1`] = `undefined`;
